middleware/bos-authentication.js

```diff
@@ -0,0 +1,326 @@
+var _ = require('lodash');
+var base64URL = require('base64url');
+
+var log;
+var loader;
+var httpMethods = ['get', 'put', 'post', 'delete', 'options', 'head', 'patch'];
+//map of middleware ids to 'true' meaning that they have been initialized
+var middlewareInitMap = {};
+
+module.exports = {
+    init : init
+};
+
+function init(app, logger, serviceLoader, swagger) {
+    log = logger;
+    loader = serviceLoader;
+    _.forEach(swagger.getSimpleSpecs(), function (api, name) {
+        var basePath = api.basePath || '';
+        /* apply security requirements to each route path*/
+        _.keys(api.paths).forEach(function (path) {
+            var pathObj = api.paths[path];
+            var routePath = basePath + _convertPathToExpress(path);
+
+            //loop for http method keys, like get an post
+            _.keys(pathObj).forEach(function (method) {
+                if (_.contains(httpMethods, method)) {
+                    var operation = pathObj[method];
+                    if (operation['security']) {
+                        operation['security'].forEach(function (securityReq) {
+                            _.forOwn(securityReq, function (scopes, securityDefn) {
+                                _applySecurityRequirement(app, method, routePath, securityDefn,
+                                    api.securityDefinitions[securityDefn],
```

Comment:
If I understand you correctly, no because all scopes should be passed into the applySecurityRequirement function. Honestly since the scopes are only used for oauth, it makes no difference at the moment.